Posix compat

ShellyHerself opened the pull request:
I still want to add a few touches after this. But this is most of it.

ShellyHerself committed:
Remove app_window reliance on PATHDIV

ShellyHerself committed:
Fix terrible default linux color picker (Depends on same name commit in binilla)

ShellyHerself committed:
Fix type in app_window

ShellyHerself committed:
Remove widgets/directory_frame reliance on PATHDIV

ShellyHerself committed:
Fix tagdependency field posix compatibility. And remove its reliance on PATHDIV.

ShellyHerself committed:
Remove windows/tools/animation_compression_window PATHDIV import to discourage its use.

ShellyHerself committed:
Remove print that shouldn't be there.

ShellyHerself committed:
Replace shitty linux filepicker with a less shitty linux filepicker.

ShellyHerself committed:
Make vim default text editor for tag scanner on linux.

ShellyHerself committed:
Correct capitalization.

ShellyHerself committed:
Fix filedialog imports.

ShellyHerself committed:
Fix property data_dir for app_window

ShellyHerself committed:
Remove windows.tools.compile_bitmap dependency on PATHDIV

ShellyHerself committed:
Switch with_suffix('') with stem

ShellyHerself committed:
Fix tag_converters posix compat.

ShellyHerself committed:
Fix not show_full

ShellyHerself committed:
Fix suffix, make some tools posix compatible.

ShellyHerself committed:
Merge remote-tracking branch 'remotes/origin/develop' into posix-compat

ShellyHerself committed:
Introduce editor constants that are posix compatible, and fully port app_window to pathlib.

ShellyHerself committed:
Make all icons use editor_constants path.

ShellyHerself committed:
Clean up miscellaneous little things I've forgotten.
I might be burning out a little.

ShellyHerself committed:
Remove editor_constants debug prints.

ShellyHerself committed:
Add print "Could not load window icon." for that condition.

ShellyHerself committed:
Fix Paths breaking config saving.

ShellyHerself committed:
Remove pointless PureWindowsPath import.

ShellyHerself committed:
unsanitize animation_compression_window

ShellyHerself committed:
Add a try except block to avoid a filepath exception breaking the tag window.

ShellyHerself committed:
Fix tag handler not giving tag windows the right tag directory.

ShellyHerself committed:
Make more tools Linux friendly.
Only dependency_window and tag_scanner to go now I think.

ShellyHerself committed:
Merge remote-tracking branch 'develop' into posix-compat

ShellyHerself committed:
Merge remote-tracking branch 'develop' into posix-compat

ShellyHerself committed:
Switch util.path references to supyr_struct

ShellyHerself committed:
Fix my utter stupidity.

ShellyHerself committed:
Shorten user directory on Linux.

ShellyHerself committed:
Fix not properly catching exceptions

ShellyHerself committed:
[directory_frame] use os.scan_dir instead of os.walk

ShellyHerself committed:
[filedialog] Replace filedialog references across the board

ShellyHerself committed:
[util] use is_in_dir function instead of try: except: pattern

ShellyHerself committed:
[directory_frame] Case insensitively sort items

ShellyHerself committed:
Merge branch 'develop' into posix-compat

ShellyHerself committed:
[field_widgets] Don't double Path convert in dependency_frame

ShellyHerself committed:
[field_widgets] DependencyFrame: Read filepath from widget...
instead of flushed copy when doing validation

ShellyHerself committed:
[field_widgets] Fix syntax error

ShellyHerself committed:
[field_widgets] DependencyFrame last few posix touchups

ShellyHerself committed:
[field_widgets] RawdataFrame(s) last few posix touchups

ShellyHerself committed:
[widgets] directory_frame, last few posix touchups

ShellyHerself committed:
[tag_converters] Last few posix touchups

ShellyHerself committed:
[tools] Clean up unused includes

ShellyHerself committed:
[app_window] Code flow clarification

ShellyHerself committed:
[global] Clean up unused includes

ShellyHerself committed:
[app_window] Clean up unused include

ShellyHerself committed:
[widgets] Fix incorrect reference

ShellyHerself committed:
[tools] Posix compat for animation compression window. Also...
Use askopenfilename instead of asksaveasfilename.

ShellyHerself committed:
[tools] Fix import in animation_compiler_window

ShellyHerself committed:
[tools] Fix typo

ShellyHerself committed:
[styles] Fix color in pop-dark style

ShellyHerself committed:
[tools] compile_bitmap posix compat

ShellyHerself committed:
[tools] Fix up tag_scanner_window for posix

ShellyHerself committed:
[tools] Fix tag dependency viewer for posix

ShellyHerself committed:
[tools] get rid of last PATHDIV usage

ShellyHerself commented on windows/tools/tag_scanner_window.py:
```diff
@@ -32,7 +32,7 @@
     # leaving this here just in case it somehow works though.
     TEXT_EDITOR_NAME = "TextEdit"
 elif platform == "linux":
```
Might not work for certain Linux versions. Needs to be
```suggestion
elif "linux" in platform:
```

ShellyHerself commented on editor_constants.py:
```diff
@@ -33,5 +36,36 @@
+
+MOZZLIB_DIR = Path(__file__).parent
+
+MOZZ_ICON_PATH = Path(MOZZLIB_DIR, "mozzarilla.ico")
+if not MOZZ_ICON_PATH.is_file():
+    MOZZ_ICON_PATH = Path(MOZZLIB_DIR, "icons", "mozzarilla.ico")
+if not MOZZ_ICON_PATH.is_file():
+    MOZZ_ICON_PATH = Path(WORKING_DIR, "mozzarilla.ico")
+if not MOZZ_ICON_PATH.is_file():
+    MOZZ_ICON_PATH = Path(WORKING_DIR, "icons", "mozzarilla.ico")
+if not MOZZ_ICON_PATH.is_file():
+    MOZZ_ICON_PATH = ""
+
+MOZZ_BITMAP_PATH = Path(MOZZLIB_DIR, "mozzarilla.png")
+if not MOZZ_BITMAP_PATH.is_file():
+    MOZZ_BITMAP_PATH = Path(MOZZLIB_DIR, "icons", "mozzarilla.png")
+if not MOZZ_BITMAP_PATH.is_file():
+    MOZZ_BITMAP_PATH = Path(WORKING_DIR, "mozzarilla.png")
+if not MOZZ_BITMAP_PATH.is_file():
+    MOZZ_BITMAP_PATH = Path(WORKING_DIR, "icons", "mozzarilla.png")
+if not MOZZ_BITMAP_PATH.is_file():
+    MOZZ_BITMAP_PATH = ""
```
If we consistently store the icons in the library folders then this is not needed.

ShellyHerself reviewed (commented):
Some of my personal notes.

MosesofEgypt committed:
Tweaked various uses of pathlib.Path objects so they work below python 3.6(os.path functions don't accept Path objects below 3.6)
Made handlers be stored by their pathlib.Path tagsdir
Fixed bugs with animation and model compiler windows not properly determining tags dir when selecting tag.
Fixed missing imports.
Fixed bitmap conveter causing index out of range exception on log creation.
Refactoring code in app_window.

MosesofEgypt committed:
Made all tools, converters, and other mozz functionality work with posix compat(hopefully?)
Made several Mozzarilla properties decorated to always be a Path object.
Removed a tags_dir property from most tools, opting to use their handler.tagsdir property directly instead(less redundancy)

MosesofEgypt committed:
Implementing some of the requested changes in pull request.
Implemented storing tags dirs(and handlers by their tags dirs) as resolved paths to ensure proper set comparison(symlinks and relative paths won't be equal until resolved).

MosesofEgypt committed:
Implementing more changes laid out in pull request comments.

ShellyHerself committed:
[CodeReview] Apply some suggestions directly

ShellyHerself committed:
[CodeReview] Implement requested changes from pullrequest

ShellyHerself committed:
[CodeReview] Fix small mistake

ShellyHerself committed:
[CodeReview] Undo renaming of config and styles paths.
Code was resorting to basing its config paths off of the ones found in 
Binilla.
The feature was basically only half implemented. >:|

ShellyHerself committed:
[app_window] Redo config path var rename

ShellyHerself committed:
[widgets] Fix rawdata widgets not being able to export sound

MosesofEgypt committed:
Fixed bug with hud_message compilation where the wrong filepath was generated for the resulting tag.
Removed debug code from compile_strings

MosesofEgypt commented on windows/tools/tag_scanner_window.py:
```diff
@@ -369,6 +358,7 @@ def scan(self):
 
                 except Exception:
                     print("    Could not scan '%s'" % tag.filepath)
+                    print(format_exc())
```
A print of the stack trace should always be before any "summary" statement. The user will see a big stack trace, and the thing they'll usually do is look at the bottom for the tl; dr

MosesofEgypt committed:
Changed how custom iconbitmap is applied for tag windows
Created custom config class with custom iconbitmap filepath
Fixed bug with converter_base not importing path_split

ShellyHerself committed:
[e_c] Remove mozzarilla sub dir from settings fir constant

ShellyHerself committed:
[widgets] case-insentively compare extension in export_node

MosesofEgypt committed:
Modified how WORKING_DIR and MOZZLIB_DIR are defined

MosesofEgypt committed:
Don't use with_suffix when it can't be guaranteed there is a suffix. Use string concatenation instead.

ShellyHerself committed:
[e_c] Fix compatibility for some path constants

ShellyHerself committed:
Merge pull request #17 from MosesofEgypt/py_35_compat

Py 35 compat

ShellyHerself committed:
[codecheck] Address pull request comments
